docs/using/troubleshooting.md

```diff
@@ -0,0 +1,310 @@
+
+If you are running a Docker container while mounting a local volume or host directory using the `-v` flag like so:
+
+```bash
+docker run -it --rm \
+    -p 8888:8888 \
+    -v <my-vol>:<container-dir> \
+    jupyter/minimal-notebook:latest
+```
+
+you might face permissions issues when trying to access the mounted volume:
+
+```bash
+# assuming we mounted the volume in /home/jovyan/stagingarea
+# root is the owner of the mounted volume
+$ ls -ld ~/stagingarea/
+drwxr-xr-x 2 root root 4096 Feb  1 12:55 stagingarea/
+
+$ touch stagingarea/kale.txt
+touch: cannot touch 'stagingarea/kale.txt': Permission denied
+```
+
+In this case, the user of the container (`jovyan`) and the owner of the mounted volume (`root`) have different permission levels and ownership over the container's directories and mounts.
+The following sections cover a few of these scenarios and how to fix them.
+
+**Some things to try:**
+
+1. **Change ownership of the volume mount**
+
+   You can change the ownership of the volume mount using the `chown` command. In the case of the docker-stacks images, you can set the `CHOWN_EXTRA` and `CHOWN_EXTRA_OPTS` environment variables.
+
+   For example, to change the ownership of the volume mount to the jovyan user (non-privileged default user in the Docker images):
+
+   ```bash
+   # running in detached mode - can also be run in interactive mode
+   docker run -d \
+       -v <my-vol>:<container-dir> \
+       -p 8888:8888 \
+       --user root \
+       -e CHOWN_EXTRA="<container-dir>" \
+       -e CHOWN_EXTRA_OPTS="-R" \
+       jupyter/minimal-notebook
+   ```
+
+   where:
+
+   - `CHOWN_EXTRA=<some-dir>`: will change the ownership and group of the specified container directory (non-recursive by default). You need to provide full paths starting with `/`.
+   - `CHOWN_EXTRA_OPTS="-R"`: will recursively change the ownership and group of the directory specified in `CHOWN_EXTRA`.
+   - `--user root`: you **must** run the container with the root user to change ownership at runtime.
+
+   now accessing the mount should work as expected:
+
+   ```bash
+   # assuming we mounted the volume in /home/jovyan/stagingarea
+   $ ls -ld ~/stagingarea
+   drwxr-xr-x 2 jovyan users 4096 Feb  1 12:55 stagingarea/
+
+   $ touch stagingarea/kale.txt
+   # jovyan is now the owner of /home/jovyan/stagingarea
+   $ ls -la ~/stagingarea/
+   -rw-r--r-- 1 jovyan users    0 Feb  1 14:41 kale.txt
+   ```
+
+   ```{admonition} Additional notes
+      - If you are mounting your volume inside the `/home/` directory, you can use the `-e CHOWN_HOME=yes` and `CHOWN_HOME_OPTS="-R"` flags instead of the `-e CHOWN_EXTRA`
+        and `-e CHOWN_EXTRA_OPTS` in the example above.
+      - This solution should work in most cases where you have created a docker volume (i.e. using the [`docker volume create --name <my-volume>`
+        command](https://docs.docker.com/storage/volumes/#create-and-manage-volumes)) and mounted it using the`-v` flag in `docker run`.
```

Comment:
I don't like this style of line breaks in markdown source code, usually, I prefer to break the lines by sentences or sentence parts - usually, it makes the source code more readable.